### PR TITLE
Fixes transformation of lowercase 1-char protein to 3-char protein and uses mocking to test

### PR DESCRIPTION
**Pull request by leoisl**
This PR has a small bugfix, but the main content is to show a way of use mocking to remove the randomness of hypothesis' `draw()` function so that we can fix what is drawn, and have deterministic tests, allowing for e.g. to test some corner cases. I am also struggling to know how to do good tests in this project, since this randomness factor seems to complicate things, and we are more interested in the strategy producing output that conforms to certain properties, rather than specific outputs. But mocking in such a way might be helpful to some tests. This is just an example, we could try this to test other parts of this function and other functions.

For full details, see https://github.com/Lab41/hypothesis-bio/issues/36

### CODE REVIEW

**Comment by vaastav** [comment]
Thanks for finding the bug!

Turns out the tests that you have written are something that can be written using minimal, thus eliminating the need for Mocking code. (Which in turn eliminates a dependency)

Can you refactor your tests based on the example here: https://github.com/Lab41/hypothesis-bio/blob/master/tests/test_rna.py

File: tests/test_protein.py [hunk 1]
```diff
@@ -47,3 +49,34 @@ def test_max_size_3_letter_abbrv():
     seq = minimal(protein(single_letter_protein=False, max_size=10))
     assert len(seq) <= 30
     assert len(seq) % 3 == 0
+
+
+# unwraps a function wrapped with the @composite decorator
+# TODO: this does not seem good and does not seem portable. Find a better way to unwrap functions out of decorators
```
Comment (mbhall88) [inline tests/test_protein.py]: What's wrong with it? Seems fine to me?

**Comment by leoisl** [comment]
I think this can be closed, as it is outdated, as https://github.com/Lab41/hypothesis-bio/commit/a2a02bcfe8fa077dafdea8565629886e03865ad7 seems to implement the changes addressed here (sorry for the slow response, I could not do it earlier). Tell me if I am wrong and should still implement the changes requested and do the PR.

**Comment by vaastav** [comment]
Yea sorry @leoisl !
I am closing this PR
